File: r/rna-transcription/rna-transcription.R
coll = 

to_rna = function(dna) {
  map = list(G = "C", C = "G", T = "A", A = "U")
  v = unlist(strsplit(toupper(dna), ""))
  
  paste(lapply(v, FUN = function(x) {
    if (is.null(map[[x]])){
      stop("invalid dna strand")
    }
    map[[x]]
  }), collapse="")
}

File: go/rna-transcription/rna_transcription.go
package strand

var MAP = map[rune]rune{
	'G': 'C',
	'C': 'G',
	'T': 'A',
	'A': 'U',
}

func ToRNA(dna string) string {

	out := make([]rune, len(dna))
	for i, c := range dna {
		out[i] = MAP[c]
	}

	return string(out)
}
